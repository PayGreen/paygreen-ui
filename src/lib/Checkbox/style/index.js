import styled from 'styled-components';
import { math } from 'polished';
import { enabled, disabled } from './base';

const checkboxSize = props =>
    props.theme.form.checkboxButtonSize[props.fieldSize];

const CheckboxBase = styled.div`
    margin: ${props => props.theme.space.sm} 0;

    input[type='checkbox'] {
        display: none;

        &:checked {
            & + label {
                &::after {
                    color: ${props => props.theme.wab.white00};
                    transform: scale(1);
                }

                &::before {
                    border-color: ${props =>
                        props.theme.color.status.success.main};
                    background-color: ${props =>
                        props.theme.color.status.success.main};
                }
                font-weight: ${props => props.theme.font.weight.bold};
            }
        }
    }

    label {
        line-height: ${checkboxSize};
        position: relative;
        padding: 0
            ${props =>
                math(
                    props.theme.form.checkboxButtonSize[props.fieldSize] +
                        '* 1.5',
                )};
        color: ${props => props.theme.wab.grey60};
        font-size: ${props => props.theme.daButton.font[props.fieldSize]};

        &::before,
        &::after {
            position: absolute;
            box-sizing: border-box;
            transition: all ${props => props.theme.transition.xs};
        }

        &::before {
            left: 0;
            top: 0;
            content: '';
            background-color: ${props => props.theme.wab.white00};
            border: solid 2px ${props => props.theme.wab.grey30};
            height: ${checkboxSize};
            width: ${checkboxSize};
            border-radius: 20%;
        }

        &::after {
            content: 'V';
            height: ${checkboxSize};
            width: ${checkboxSize};
            display: flex;
            justify-content: center;
            align-items: center;
            color: ${props => props.theme.wab.white00};
            left: 0;
            top: 0;
            transform: scale(0);
        }
    }

    ${props => (props.checkboxDisabled ? disabled : enabled)};
`;

export { CheckboxBase };
