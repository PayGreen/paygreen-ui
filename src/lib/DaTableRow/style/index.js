import styled from 'styled-components';
import { gridTemplate } from './constants';
import { cellsAreas, toggableStyle, activeStyle, hoverStyle } from './base';

const DaTableRowBase = styled.div`
    transition: all ${props => props.theme.transition.sm};
    ${hoverStyle};
    ${props => (props.isActive ? activeStyle : null)};

    @media (${props => props.theme.query.max.lg}) {
        margin: ${props => props.theme.space.xs};
        padding: ${props => props.theme.space.xs};
        border-radius: ${props => props.theme.radius.sm};
        display: grid;
        grid-template-areas: ${props =>
            gridTemplate(
                props.mainCellCount,
                props.notMainCellCount,
                props.hasCheckbox,
            )};
        grid-template-columns: ${props =>
            props.hasCheckbox
                ? 'repeat(' +
                  parseInt(props.mainCellCount - 1) +
                  ', auto) min-content'
                : 'repeat(' + props.mainCellCount + ', auto)'};

        ${cellsAreas};
        ${toggableStyle};
    }

    @media (${props => props.theme.query.min.lg}) {
        display: table-row;

        .cell-checkbox {
            padding-right: 0;
        }
    }
`;

export { DaTableRowBase };
