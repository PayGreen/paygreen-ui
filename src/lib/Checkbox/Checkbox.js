import React from 'react';
import PropTypes from 'prop-types';
import { buttonSizeOptions, buttonSizeDefault } from '../../shared/constants';
import { CheckboxBase } from './style';
import { CheckBoldIcon } from '../Icon/Icon';

const Checkbox = props => {
    const { fieldSize, label, id, inputRef, ...rest } = props;

    return (
        <CheckboxBase
            theme={props.theme} // not necessary, only needed for tests
            fieldSize={fieldSize}
            isDisabled={props.disabled}
            hasLabel={label && label.length > 0}
        >
            <input {...rest} ref={inputRef} id={id} type="checkbox" />

            <label htmlFor={id}>
                <CheckBoldIcon
                    theme={props.theme} // not necessary, only needed for tests
                />

                {label}
            </label>
        </CheckboxBase>
    );
};

Checkbox.propTypes = {
    id: PropTypes.string.isRequired,
    label: PropTypes.string,
    disabled: PropTypes.bool,
    fieldSize: PropTypes.oneOf(Object.values(buttonSizeOptions)),
    inputRef: PropTypes.oneOfType([
        PropTypes.func,
        PropTypes.shape({ current: PropTypes.any }),
    ]),
};

Checkbox.defaultProps = {
    disabled: false,
    fieldSize: buttonSizeDefault,
};

export default Checkbox;
