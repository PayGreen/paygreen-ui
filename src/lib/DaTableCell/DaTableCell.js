import React from 'react';
import PropTypes from 'prop-types';
import { DaTableCellBase } from './style';

const DaTableCell = props => {
    const label = props.label ? (
        <span className="cell-label">{props.label}</span>
    ) : null;

    const content = props.children ? (
        <span className="cell-content">{props.children}</span>
    ) : (
        <>&nbsp;</>
    );

    return (
        <DaTableCellBase {...props} hasLabel={label ? true : false}>
            {label}
            {content}
            {props.field ? props.field : null}
        </DaTableCellBase>
    );
};

DaTableCell.propTypes = {
    isMain: PropTypes.bool,
    isCheckbox: PropTypes.bool,
    isId: PropTypes.bool,
    label: PropTypes.string,
    field: PropTypes.element,
};

DaTableCell.defaultProps = {
    isMain: true,
    isCheckbox: false,
    isId: false,
};

export default DaTableCell;
