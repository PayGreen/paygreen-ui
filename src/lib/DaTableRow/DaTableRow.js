import React from 'react';
import PropTypes from 'prop-types';
import {

} from '../../shared/constants';
import { DaTableRowBase } from './style';

const DaTableRow = (props) => {
    return <DaTableRowBase
        {...props}
    >
        {props.children}
    </DaTableRowBase>;
};

DaTableRow.propTypes = {

};

DaTableRow.defaultProps = {

};

export default DaTableRow;