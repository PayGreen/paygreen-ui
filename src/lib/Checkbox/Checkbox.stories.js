import React from 'react';
import {
    folder,
    buttonSizeOptions,
    buttonSizeDefault,
} from '../../shared/constants';
import labels from '../../shared/labels';
import Checkbox from './Checkbox';

export default {
    title: folder.form + folder.sub.checkbox + 'Checkbox',
    argTypes: {
        disabled: {
            name: labels.disabled,
            control: 'boolean',
        },
        fieldSize: {
            name: labels.fieldSize,
            options: Object.values(buttonSizeOptions),
            control: 'radio',
        },
    },
};

export const CheckboxStory = ({ ...args }) => (
    <Checkbox id="first" label="First Choice" {...args} />
);

CheckboxStory.args = {
    fieldSize: buttonSizeDefault,
    disabled: false,
};
