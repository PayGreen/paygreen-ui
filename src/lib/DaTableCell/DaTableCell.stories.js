import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, text } from '@storybook/addon-knobs';
import { folder } from '../../shared/constants';
import DaTableCell from './DaTableCell';

storiesOf(folder.tables + folder.sub.daTable + 'DaTableCell', module)
    .addDecorator(withKnobs)
    .add('DaTableCell', () => (
        <DaTableCell
            label={text('Label', 'Label')}
            isMain={boolean('Is main', true)}
            isId={boolean('Is id', false)}
        >
            Sample
        </DaTableCell>
    ));
