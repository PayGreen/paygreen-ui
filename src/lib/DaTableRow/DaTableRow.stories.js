import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { folder } from '../../shared/constants';
import DaTableCell from '../DaTableCell/DaTableCell';
import Checkbox from '../Checkbox/Checkbox';
import DaTableRow from './DaTableRow';

storiesOf(folder.table + folder.sub.daTable + 'DaTableRow', module)
    .addDecorator(withKnobs)
    .add('DaTableRow', () => {
        const [isActive, setActive] = useState(false);

        return (
            <DaTableRow isActive={isActive}>
                <DaTableCell>
                    <Checkbox
                        id="select"
                        onChange={() => setActive(!isActive)}
                        checked={isActive}
                    />
                </DaTableCell>

                <DaTableCell isId={true}>3456</DaTableCell>

                <DaTableCell isMain={false} label="Date">
                    27/05/2020
                </DaTableCell>

                <DaTableCell>Marie Perez</DaTableCell>

                <DaTableCell>34.56&nbsp;â‚¬</DaTableCell>

                <DaTableCell isMain={false} label="Type">
                    Cash
                </DaTableCell>

                <DaTableCell isMain={false} label="A very long label">
                    <span>A very long content without bold</span>
                </DaTableCell>
            </DaTableRow>
        );
    });
