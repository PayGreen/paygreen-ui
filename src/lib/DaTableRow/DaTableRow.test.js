import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import Checkbox from '../Checkbox/Checkbox';
import DaTableCell from '../DaTableCell/DaTableCell';
import DaTableRow from './DaTableRow';

it('renders without crashing', () => {
    const component = TestRenderer.create(
        <DaTableRow theme={ThemeDefault}>
            <DaTableCell theme={ThemeDefault}>
                <Checkbox theme={ThemeDefault} id="select"/>
            </DaTableCell>

            <DaTableCell theme={ThemeDefault} isId={true}>
                4217
            </DaTableCell>

            <DaTableCell theme={ThemeDefault}>Something</DaTableCell>

            <DaTableCell theme={ThemeDefault} isMain={false}>
                Something else
            </DaTableCell>
        </DaTableRow>,
    );
    expect(component.toJSON()).toMatchSnapshot();
});
