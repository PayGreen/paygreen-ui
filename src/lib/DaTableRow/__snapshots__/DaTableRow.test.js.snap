// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="sc-bwzfXH fmyGrX"
>
  <div
    className="sc-bdVaJa igqJNM cell-checkbox"
  >
    <span
      className="cell-content"
    >
      <input
        type="checkbox"
      />
    </span>
  </div>
  <div
    className="sc-bdVaJa bSlSYA cell-main cell-main-1"
  >
    <span
      className="cell-content"
    >
      4217
    </span>
  </div>
  <div
    className="sc-bdVaJa KnVGS cell-main cell-main-2"
  >
    <span
      className="cell-content"
    >
      Something
    </span>
  </div>
  <div
    className="sc-bdVaJa iyIPAb cell-basic cell-basic-1"
  >
    <span
      className="cell-content"
    >
      Something else
    </span>
  </div>
</div>
`;
